### ISSUE TITLE
seeking in streaming write file should be an error

### ISSUE THREAD

**Issue by christopherhesse** [christopher-hesse/blobfile#163]
Something like this, should also update the comment about `_offset` and add a test.

```
    def seek(self, offset: int, whence: int = io.SEEK_SET) -> int:
        if whence == io.SEEK_SET:
            new_offset = offset
        elif whence == io.SEEK_CUR:
            new_offset = self._offset + offset
        elif whence == io.SEEK_END:
            new_offset = self._size + offset
        else:
            raise ValueError(
                f"Invalid whence ({whence}, should be {io.SEEK_SET}, {io.SEEK_CUR}, or {io.SEEK_END})"
            )
        if new_offset != self._offset:
            raise io.UnsupportedOperation("cannot seek streaming write files")
        return self._offset

    def truncate(self, size: Optional[int]=None) -> int:
        raise io.UnsupportedOperation("not truncateable")

    def tell(self) -> int:
        return self._offset + len(self._buf)```

**Comment by christopher-hesse** [comment]
@unixpickle actually `tell()` was the only bug I found here.  `f.seek()` correctly returns an error and fixing `tell()` appears to be sufficient to fix the zipfile issue, it must have some other mechanism to deal with unseekable files.

**Comment by christopher-hesse** [comment]
Fixed in 1.1.2
